pulsar-client-admin-api/pom.xml

```diff
@@ -36,13 +36,32 @@
 
         <dependency>
             <groupId>${project.groupId}</groupId>
-            <artifactId>pulsar-common</artifactId>
-            <version>${project.parent.version}</version>
+            <artifactId>pulsar-client-api</artifactId>
+            <version>${project.version}</version>
         </dependency>
 
         <dependency>
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>io.swagger</groupId>
+            <artifactId>swagger-annotations</artifactId>
+            <scope>provided</scope>
```

Comment:
I'm ok if this was a step of the overall changes, though in the final form we cannot have these provided dependencies. This just removes the dependencies for the user, but then the error will be at runtime, especially since these symbols will be then shaded.

Reply:
Since we cannot take out the Swagger annotations, that was the reason I was suggesting to convert the POJO classes into interfaces and have the Swagger annotations on the implementation only.

Reply:
@merlimat thanks for your comments, I have removed swagger dependency from `pulsar-client-admin-api` here and convert related classes into interfaces, PTAL, thanks.